src/foam/lib/PropertyPredicate.js

```diff
@@ -0,0 +1,26 @@
+foam.INTERFACE({
+  package: 'foam.lib',
+  name: 'PropertyPredicate',
+
+  methods: [
+    {
+      name: 'propertyPredicateCheck',
+      type: 'boolean',
+      args: [
+        {
+          name: 'x',
+          type: 'foam.core.X'
+        },
+        {
+          name: 'fo',
+          type: 'FObject'
+        },
+        {
+          name: 'prop',
+          type: 'foam.core.PropertyInfo'
```

Comment:
`foam.core.PropertyInfo` is not a modelled object. It only exists for java so use `javaType` instead of `type`.